packages/rum/test/specs/apm-base.spec.js

```diff
@@ -36,17 +36,41 @@ import {
 import { TRANSACTION_END } from '@elastic/apm-rum-core/src/common/constants'
 import { getGlobalConfig } from '../../../../dev-utils/test-config'
 import Promise from 'promise-polyfill'
+import {
+  hidePageSynthetically,
+  setDocumentVisibilityState
+} from '@elastic/apm-rum-core/test'
```

Comment:
does this work? now the test of packages/rum should build packages/rum-core each time for tests, I dont think this would work with current build system. We only build on packages/rum/src changes.

Reply:
Thanks for asking this! When I was doing this that was precisely my doubt, it felt a bit weird to import something from another package, but at that moment it's when I saw this line in the very same file:

```import { TRANSACTION_END } from '@elastic/apm-rum-core/src/common/constants'```. It turns out that is something that we are already doing. And when I started looking at other specs I saw that it's our way of doing across all the project, examples:

- https://github.com/elastic/apm-agent-rum-js/blob/main/packages/rum/test/specs/index.spec.js
- https://github.com/elastic/apm-agent-rum-js/blob/main/packages/rum-vue/test/specs/index.spec.js
- https://github.com/elastic/apm-agent-rum-js/blob/main/packages/rum-react/test/specs/get-with-transaction.spec.js
- and many more

And yes, it works. The current build system relies on `lerna` for this. For instance, when you execute the `npm run build` corresponding to the main package.json, the order of builds is going to be the following one:

- apm-rum-core
- apm-rum
- apm specific frameworks

This is happening because in the package.json of each package we are saying which packages depend on each other:

- **apm-rum** depends on **apm-rum-core**
- **apm-rum-react** depends on **@elastic/apm-rum**
- **apm-rum-vue** and **apm-rum-angular**  depends on **@elastic/apm-rum** and **@elastic/apm-rum-core**

For instance, **lerna** guarantees that the build for **apm-rum** will not start until the one from **apm-rum-core** is finished, since it's a dependency.

Hope this clarify your question.

--
Having said that, in future projects where we could use lerna with multiple packages, we should think if this is the way to proceed since now we the packages are not really independent. We will trade-offs on those discussions :)

Reply:
> import { TRANSACTION_END } from '@elastic/apm-rum-core/src/common/constants'

Was pretty sure this would work as we build `src` all the time and As you mentioned lerna does guarantee the order based on the dependency tree. But did not knew `test` directory was also taken in to account which feels a bit wrong. But all good 👍🏽